[BUG] mastracode: hooks.json commands cannot use quoted paths on Windows (cmd spawned without windowsVerbatimArguments)

### Describe the Bug

## Problem

On Windows, a Mastra Code hook whose `command` contains a **quoted path** never runs. The quotes reach the child process literally.

Given `~/.mastracode/hooks.json`:

```json
{
  "SessionStart": [
    { "type": "command", "command": "node \"C:/tools/my-hook.mjs\"" }
  ]
}
```

the hook fails with:

```
Error: Cannot find module 'C:\some\cwd\"C:\tools\my-hook.mjs"'
```

Note the literal `"` characters embedded in the resolved path.

Because unquoted paths split on spaces, the combined effect is that **a hook command path containing spaces cannot be expressed at all** on Windows — quoting breaks it, and not quoting breaks it. This affects any user whose paths contain a space (e.g. `C:\Users\First Last\...`, or anything under `Documents`/`Program Files`).

This is also easy to miss: a `PostToolUse`/`SessionStart` hook that fails this way is reported only as a warning, so it looks like the hook is configured but does nothing.

## Root cause

`mastracode/sdk/src/hooks/executor.ts` spawns the shell without `windowsVerbatimArguments`:

```ts
const isWindows = process.platform === 'win32';
const shell = isWindows ? 'cmd' : '/bin/sh';
const shellArgs = isWindows ? ['/c', hook.command] : ['-c', hook.command];

const child = spawn(shell, shellArgs, {
  stdio: ['pipe', 'pipe', 'pipe'],
  cwd: stdinPayload.cwd,
  env: { ...process.env, MASTRA_HOOK_EVENT: stdinPayload.hook_event_name },
});
```

When `spawn` is given `cmd` as the file (rather than `shell: true`), Node applies its normal Windows argument escaping to `hook.command`, so the inner `"` are escaped and `cmd.exe` passes them through to the child as literal characters.

Same class of bug as #17646 (Windows shell-quote escaping in `create-mastra`), different package.

## Reproduction

Requires only Node on Windows. Mirrors `executeHook` exactly:

```js
import { spawn } from 'node:child_process';
import fs from 'node:fs';
import os from 'node:os';
import path from 'node:path';

const dir = fs.mkdtempSync(path.join(os.tmpdir(), 'mc-hook-repro-'));
const plain = path.join(dir, 'hook.mjs');
const spacedDir = path.join(dir, 'dir with spaces');
fs.mkdirSync(spacedDir);
const spaced = path.join(spacedDir, 'hook.mjs');
fs.writeFileSync(plain, 'process.stdout.write("HOOK_RAN");\n');
fs.writeFileSync(spaced, 'process.stdout.write("HOOK_RAN");\n');

function runHook(command, extraOpts = {}) {
  return new Promise(resolve => {
    const isWindows = process.platform === 'win32';
    const child = spawn(isWindows ? 'cmd' : '/bin/sh', isWindows ? ['/c', command] : ['-c', command], {
      stdio: ['pipe', 'pipe', 'pipe'],
      ...extraOpts,
    });
    let out = '', err = '';
    child.stdout.on('data', d => (out += d));
    child.stderr.on('data', d => (err += d));
    child.on('close', code => resolve({ code, out: out.trim(), err: err.trim() }));
    child.stdin.end();
  });
}

const p = plain.replace(/\\/g, '/');
const s = spaced.replace(/\\/g, '/');

const cases = [
  ['unquoted, no spaces in path', `node ${p}`, {}],
  ['quoted, no spaces in path', `node "${p}"`, {}],
  ['quoted, path contains spaces', `node "${s}"`, {}],
  ['unquoted, path contains spaces', `node ${s}`, {}],
  ['quoted + windowsVerbatimArguments (proposed fix)', `node "${s}"`, { windowsVerbatimArguments: true }],
];

console.log(`platform: ${process.platform}   node: ${process.version}\n`);
for (const [label, command, opts] of cases) {
  const r = await runHook(command, opts);
  const ok = r.out === 'HOOK_RAN';
  console.log(`${ok ? 'PASS' : 'FAIL'}  ${label}`);
  if (!ok) {
    const line = r.err.split('\n').find(l => l.includes('Error')) || r.err.split('\n')[0] || '(no stderr)';
    console.log(`      exit=${r.code}  ${line.trim().slice(0, 120)}`);
  }
}
fs.rmSync(dir, { recursive: true, force: true });
```

### Actual output

```
platform: win32   node: v24.16.0

PASS  unquoted, no spaces in path
FAIL  quoted, no spaces in path
      exit=1  Error: Cannot find module 'C:\...\"C:\Users\...\Temp\mc-hook-repro-3ytfmv\hook.mjs"'
FAIL  quoted, path contains spaces
      exit=1  Error: Cannot find module 'C:\...\"C:\Users\...\Temp\mc-hook-repro-3ytfmv\dir'
FAIL  unquoted, path contains spaces
      exit=1  Error: Cannot find module 'C:\Users\...\Temp\mc-hook-repro-3ytfmv\dir'
PASS  quoted + windowsVerbatimArguments (proposed fix)
```

### Expected behaviour

A quoted path in a hook `command` should run, and a path containing spaces should be expressible — as it is on macOS/Linux via `/bin/sh -c`.

### Proposed fix

Pass `windowsVerbatimArguments: true` when spawning on Windows, so `cmd.exe` receives the command string verbatim:

```ts
const child = spawn(shell, shellArgs, {
  stdio: ['pipe', 'pipe', 'pipe'],
  cwd: stdinPayload.cwd,
  ...(isWindows ? { windowsVerbatimArguments: true } : {}),
  env: { ...process.env, MASTRA_HOOK_EVENT: stdinPayload.hook_event_name },
});
```

The last repro case confirms this makes both quoted and space-containing paths work. Happy to open a PR with the fix plus a regression test if that's useful.

### Environment

- `mastracode` 0.32.4, `@mastra/code-sdk` 1.1.1 (current npm latest)
- Also present on the current default branch: `mastracode/sdk/src/hooks/executor.ts`
- Windows 11, Node v24.16.0

--- a/mastracode/sdk/src/hooks/executor.test.ts
+++ b/mastracode/sdk/src/hooks/executor.test.ts
@@ -1,0 +1,74 @@
+import { EventEmitter } from 'node:events';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+import type { HookDefinition, HookStdinSession } from './types.js';
+
+const { spawnMock } = vi.hoisted(() => ({
+  spawnMock: vi.fn(),
+}));
+
+vi.mock('node:child_process', () => ({
+  spawn: spawnMock,
+}));
+
+const { executeHook } = await import('./executor.js');
+
+class FakeChildProcess extends EventEmitter {
+  stdout = new EventEmitter();
+  stderr = new EventEmitter();
+  stdin = { write: vi.fn(), end: vi.fn() };
+  kill = vi.fn();
+}
+
+function makeHook(command: string): HookDefinition {
+  return { type: 'command', command };
+}
+
+function makeStdin(): HookStdinSession {
+  return { session_id: 'session-1', cwd: '/tmp/project', hook_event_name: 'SessionStart' };
+}
+
+describe('executeHook Windows argument handling', () => {
+  let originalPlatform: NodeJS.Platform;
+
+  beforeEach(() => {
+    spawnMock.mockReset();
+    originalPlatform = process.platform;
+  });
+
+  afterEach(() => {
+    Object.defineProperty(process, 'platform', { value: originalPlatform, configurable: true });
+  });
+
+  it('passes windowsVerbatimArguments: true when spawning cmd on win32', async () => {
+    Object.defineProperty(process, 'platform', { value: 'win32', configurable: true });
+
+    const fakeChild = new FakeChildProcess();
+    spawnMock.mockImplementation(() => {
+      setImmediate(() => fakeChild.emit('close', 0));
+      return fakeChild;
+    });
+
+    await executeHook(makeHook('node "C:/tools/my-hook.mjs"'), makeStdin());
+
+    expect(spawnMock).toHaveBeenCalledWith(
+      'cmd',
+      ['/c', 'node "C:/tools/my-hook.mjs"'],
+      expect.objectContaining({ windowsVerbatimArguments: true }),
+    );
+  });
+
+  it('does not pass windowsVerbatimArguments on non-Windows platforms', async () => {
+    Object.defineProperty(process, 'platform', { value: 'darwin', configurable: true });
+
+    const fakeChild = new FakeChildProcess();
+    spawnMock.mockImplementation(() => {
+      setImmediate(() => fakeChild.emit('close', 0));
+      return fakeChild;
+    });
+
+    await executeHook(makeHook('node "/tools/my-hook.mjs"'), makeStdin());
+
+    const callOptions = spawnMock.mock.calls[0]?.[2] as Record<string, unknown>;
+    expect(callOptions).not.toHaveProperty('windowsVerbatimArguments');
+  });
+});

--- a/mastracode/sdk/src/hooks/executor.ts
+++ b/mastracode/sdk/src/hooks/executor.ts
@@ -21,6 +21,12 @@ export async function executeHook(hook: HookDefinition, stdinPayload: HookStdin)
     const child = spawn(shell, shellArgs, {
       stdio: ['pipe', 'pipe', 'pipe'],
       cwd: stdinPayload.cwd,
+      // Without this, Node's default Windows argument escaping re-quotes
+      // hook.command before cmd.exe ever sees it, so a quoted path (or any
+      // path containing spaces) reaches the child process with literal "
+      // characters embedded in it. mastracode/tui's shell-runner.ts already
+      // sets this for its own cmd invocations; this call site just missed it.
+      ...(isWindows ? { windowsVerbatimArguments: true } : {}),
       env: {
         ...process.env,
         MASTRA_HOOK_EVENT: stdinPayload.hook_event_name,

